### DISCUSSION TITLE
MacOS编译错误：Package minted Error: You must have `pygmentize' installed to use this package

### DISCUSSION THREAD

**Discussion by Wizard23333** [TJ-CSCCG/TongjiThesis#26] [Q&A] [answered]
### 使用平台详情

* [ ] 线上：Overleaf
* [x] 本地：MacOS
* [ ] 本地：Windows
* [ ] 本地：各种 Linux 发行版
* [ ] 不，我使用的是 _________

### 编译日志

<details><summary><b>报错内容</b> <i>[点击打开]</i></summary>
<div>

```log
tongji-undergrad-thesis/main.tex:21: Package minted Error: You must have `pygmentize' installed to use this package.
```

</div>
</details>

### 问题描述

在macos上编译发现的一个问题，参考解决方案：https://tex.stackexchange.com/questions/48018/minted-not-working-on-mac
![截屏2023-04-24 17 45 57](https://user-images.githubusercontent.com/55119072/233961429-bda939af-c932-4c8f-a000-43d2f1f1d2df.png)

**Comment by RizhongLin** [comment]
That's correct! To utilize the `minted` package, you'll need to first install Python and then `pygmentize`. We'll be sure to revise the README once we've addressed the current bugs that we've identified. 

Please note that the Overleaf project is integrated with `minted`, so you won't need to worry about installing anything if you're using Overleaf. However, if you're working on the project locally, you'll need to follow the installation steps.

**Comment by RizhongLin** [comment]
I understand what you're saying. Could you share with me the version of macOS you're currently using, the version of TeX Live you have installed, and the method you employed to install `pygmentize`?

**Reply by Wizard23333** [reply]
我认为这个问题可能是在macos上使用这个模版的一个常见问题，解决这个问题花费了我几个小时，所以我希望列举出来供其他遇到相似问题的人查看
pygmentize 安装方法：pip3 install pygments

macos version: 12.6.3
tex live version: TeX 3.141592653 (TeX Live 2023)
mactex version: 15 March 2023

**Reply by RizhongLin** [reply, marked as answer]
你好，

谢谢你能够提出这个 issue，让我们了解到文档中缺少了 `pygmentize` 的相关说明。

但我认为这并不是一个常见的问题，因为我和 @skyleaworlder  在我们的 Macbook 上都没有遇到过这个问题，编译都非常顺利。问题可能出在你没有将 Python 或 `pygmentize` 的路径加入环境变量 `PATH` 中，导致与 `minted` 相关的部分无法编译。

你所提到的解决方法：

>  I just encountered the same issue and copying pygmentize from `/Library/Frameworks/Python.framework/Versions/2.6/bin/pygmentize` to `/usr/local/bin` resolved the problem for me.

其实也是把 `pygmentize` 的安装路径移到了已经包括在环境变量的路径中（`/usr/local/bin` 默认是包括在环境变量中的！），这样系统才能找到并正确使用 `pygmentize`，从而解决了问题。

我们已经在 README.md 中更新了关于安装 Python 和 `pygmentize` 相关步骤的说明。

@skyleaworlder 之前回复你时也已经提到了这个问题：

> I think I understand what you mean, but personally I don't think this is a bug.
> 
> The minted package does indirectly use `pygmentize`, but failing to properly install or locate `pygmentize` on the computer shouldn't be considered an error of this template. However, I agree with @tobiasforest 's approach that this is a `docs` issue, as our template lacks explanation for minted usage.
> 
> Please let me know if I understood the issue and your intended meaning correctly.

再次感谢你提出这个 issue！谢谢你对我们的支持！

**Reply by skyleaworlder** [reply]
我仍然认为这是一个基本的 pip 问题，本 discussion 和这个 https://stackoverflow.com/questions/35898734/pip-installs-packages-successfully-but-executables-not-found-from-command-line  其实是一件事。

但同时我也认为确实需要一些文档来说明。另外我感觉可以效仿 sjtug 在本仓库写一些 wiki，告知用户如何正确使用 pip 。这些本来并不应占据 readme 的篇幅。

**Comment by skyleaworlder** [comment]
I think I understand what you mean, but personally I don't think this is a bug.

The minted package does indirectly use `pygmentize`, but failing to properly install or locate `pygmentize` on the computer shouldn't be considered an error of this template. However, I agree with @tobiasforest 's approach that this is a `docs` issue, as our template lacks explanation for minted usage.

Please let me know if I understood the issue and your intended meaning correctly.
